packages/flutter_tools/lib/src/android/java.dart

```diff
@@ -0,0 +1,55 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../base/file_system.dart';
+import '../base/os.dart';
+import '../base/platform.dart';
+import 'android_studio.dart';
+
+const String _javaHomeEnvironmentVariable = 'JAVA_HOME';
+const String _javaExecutable = 'java';
+
+/// Attempts to find the home directory of the local Java installation used
```

Comment:
nit here and for `findJavaBinary` separate the first sentence of a dartdoc into its own paragraph: https://dart.dev/guides/language/effective-dart/documentation#do-separate-the-first-sentence-of-a-doc-comment-into-its-own-paragraph